marin/execution/executor.py

```diff
@@ -185,10 +185,11 @@ class InputName:
 
     step: ExecutorStep | None
     name: str | None
-    should_skip_parent: bool = False
+    wait_on_step: bool = True
```

Comment:
why don't we call this nonblocking just to have one name instead of two?

Reply:
well, we can't have a method named nonblocking and a member named nonblocking

Reply:
Hmm...with_nonblocking() or set_nonblocking() (don't love it)?  or `_nonblocking` (since people don't access this typically)?